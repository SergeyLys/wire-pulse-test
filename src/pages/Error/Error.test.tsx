import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as Router from 'react-router-dom';
import ErrorPage from './Error';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteError: jest.fn()
}));

describe('ErrorPage component', () => {
  it('renders the error message', () => {
    jest.spyOn(Router, 'useRouteError').mockReturnValue({ statusText: 'Not Found' });

    render(<ErrorPage />);

    const errorMessage = screen.getByText('Not Found');
    expect(errorMessage).toBeInTheDocument();

    const tree = renderer.create(<ErrorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a default error message when no error is provided', () => {
    jest.spyOn(Router, 'useRouteError').mockReturnValue(null);

    render(<ErrorPage />);

    const defaultErrorMessage = screen.getByText('Sorry, an unexpected error has occurred.');
    expect(defaultErrorMessage).toBeInTheDocument();

    const tree = renderer.create(<ErrorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
