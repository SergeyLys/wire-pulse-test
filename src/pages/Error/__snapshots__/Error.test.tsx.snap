// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component renders a default error message when no error is provided 1`] = `
<div
  className="flex align-center justify-center flex-column flex-grow-1"
>
  <div
    className="error-container"
  >
    <h1
      className="error-title"
    >
      Oops!
    </h1>
    <p
      className="error-subtitle"
    >
      Sorry, an unexpected error has occurred.
    </p>
  </div>
</div>
`;

exports[`ErrorPage component renders the error message 1`] = `
<div
  className="flex align-center justify-center flex-column flex-grow-1"
>
  <div
    className="error-container"
  >
    <h1
      className="error-title"
    >
      Oops!
    </h1>
    <p
      className="error-subtitle"
    >
      Sorry, an unexpected error has occurred.
    </p>
    <p
      className="error-text"
    >
      <i>
        Not Found
      </i>
    </p>
  </div>
</div>
`;
